app/notification.go

```diff
@@ -742,3 +735,48 @@ func GetMentionKeywordsInChannel(profiles map[string]*model.User) map[string][]s
 
 	return keywords
 }
+
+func ShouldSendPushNotification(user *model.User, channelNotifyProps model.StringMap, wasMentioned bool, status *model.Status, post *model.Post) bool {
+	return DoesNotifyPropsAllowPushNotification(user, channelNotifyProps, post, wasMentioned) &&
+		DoesStatusAllowPushNotification(user.NotifyProps, status, post.ChannelId)
+}
+
+func DoesNotifyPropsAllowPushNotification(user *model.User, channelNotifyProps model.StringMap, post *model.Post, wasMentioned bool) bool {
```

Comment:
I don't have confidence in this function. Can we write a unit test to test every case so we know it works?